app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -77,24 +82,60 @@ class AuditController @Inject() (
 
   private val predicate = Predicate.Permission(Resource(ResourceType("transit-movements-auditing"), ResourceLocation("audit")), IAAction("WRITE"))
 
-  def post(auditType: AuditType): Action[Source[ByteString, _]] = internalAuth(predicate).streamFromFile {
-
-    implicit request =>
-      if (appConfig.auditingEnabled) {
-        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-
-        (for {
-          stream <- getSource(auditType, request)(exceedsMessageSize)
-          result <- auditService.send(auditType, stream).asPresentation
-        } yield result)
-          .fold(
-            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-            _ => Accepted
-          )
-      } else {
-        Future.successful(Accepted)
+  def post(auditType: AuditType): Action[Source[ByteString, _]] =
+    internalAuth(predicate).streamFromFile {
+      implicit request =>
+        if (auditType.messageType.isDefined) postMessageTypeAudit(auditType)
+        else postStatusAudit(auditType)
+    }
+
+  def postMessageTypeAudit(auditType: AuditType)(implicit request: Request[Source[ByteString, _]]): Future[Result] =
+    if (appConfig.auditingEnabled) {
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+
+      (for {
+        stream <- getSource(auditType, request)(exceedsMessageSize)
+        result <- auditService.send(auditType, stream).asPresentation
+      } yield result)
+        .fold(
+          presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+          _ => Accepted
+        )
+    } else {
+      Future.successful(Accepted)
+    }
+
+  def postStatusAudit(auditType: AuditType)(implicit request: Request[Source[ByteString, _]]): Future[Result] =
+    (for {
+      string  <- extractBody(request.body)
+      details <- parseDetails(string)
```

Comment:
Here after extraction we need to send data to audit service?

Reply:
Yes, but that's a separate ticket. This ticket was only meant to separate the paths so work can be done in parallel beyond this, so for this ticket, this is okay.